cli/src/cli.ts

```diff
@@ -1,22 +1,22 @@
 import { Range } from 'immutable'
+import fs from 'node:fs/promises'
 
-import type { TrainerLog, data, Task } from '@epfml/discojs-core'
+import type { data, Task } from '@epfml/discojs-core'
 import { Disco, aggregator as aggregators, client as clients } from '@epfml/discojs-core'
 import { startServer } from '@epfml/disco-server'
 
-import { saveLog } from './utils.js'
 import { getTaskData } from './data.js'
 import { args } from './args.js'
 
-async function runUser (task: Task, url: URL, data: data.DataSplit): Promise<TrainerLog> {
+async function runUser (task: Task, url: URL, data: data.DataSplit): Promise<unknown> {
   const client = new clients.federated.FederatedClient(url, task, new aggregators.MeanAggregator())
 
   // force the federated scheme
   const disco = new Disco(task, { scheme: 'federated', client })
 
-  await disco.fit(data)
-  await disco.close()
-  return await disco.logs()
+  const logs = await disco.fitted(data)
```

Comment:
I wanted to potentially simplify the use of `Disco.fit`, thinking that awaiting a value rather than using a generator function would be easier. but yeah, I don't like it either, one can either `for await (const _ of disco.fit);` or get the logs themself. getting rid of it